doc/descriptors.md

```diff
@@ -0,0 +1,95 @@
+# Support for Output Descriptors in Bitcoin Core
+
+Since Bitcoin Core v0.17, there is support for Output Descriptors in the
+`scantxoutset` RPC call. This is a simple language which can be used to
+describe collections of output scripts.
+
+This document describes the language. For the specifics on usage for scanning
+the UTXO set, see the `scantxoutset` RPC help.
+
+## Features
+
+Output descriptors currently support:
+- Pay-to-pubkey scripts (P2PK), through the `pk` function.
+- Pay-to-pubkey-hash scripts (P2PK), through the `pkh` function.
+- Pay-to-witness-pubkey-hash scripts (P2WPK), through the `wpkh` function.
+- Pay-to-script-hash scripts (P2SH), through the `sh` function.
+- Pay-to-witness-script-hash scripts (P2WSH), through the `wsh` function.
+- Multisig scripts, through the `multi` function.
+- Any type of supported address (P2PKH, P2SH, Bech32), through the `addr` function.
+- Raw hex scripts through the `raw` function.
+- Public keys (compressed and uncompressed) in hex notation, or BIP32 extended pubkeys with derivation path.
+
+## Examples
+
+A P2PK descriptor with a fixed public key:
+- `pk(0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798)`
+
+A descriptor for a P2SH-embedded witness output with a fixed public key:
+- `sh(wpkh(02e493dbf1c10d80f3581e4904930b1404cc6c13900ee0758474fa94abe8c4cd13))`
+
+A native segwit 1-of-2 multisig descriptor:
+- `wsh(multi(1,022f8bde4d1a07209355b4a7250a5c5128e88b84bddc619ab7cba8d569b240efe4,025cbdf0646e5db4eaa398f365f2ea7a0e3d419b7e0330e39ce92bddedcac4f9bc))`
+
+A chain of P2PKH outputs:
+- `pkh(xpub68Gmy5EdvgibQVfPdqkBBCHxA5htiqg55crXYuXoQRKfDBFA1WEjWgP6LHhwBZeNK1VTsfTFUHCdrfp1bgwQ9xv5ski8PX9rL2dZXvgGDnw/1/0/*)`
+
+## Quick reference
+
+Scripts:
+- `sh(SCRIPT)` (top level only): P2SH embed the argument.
+- `wsh(SCRIPT)` (not inside another 'wsh'): P2WSH embed the argument.
+- `pk(PUBKEY)` (anywhere): P2PK output for the given public key.
+- `pkh(PUBKEY)` (anywhere): P2PKH output for the given public key (use `addr` if you only know the pubkey hash).
+- `wpkh(PUBKEY)` (not inside `wsh`): P2WPKH output for the given compressed pubkey.
+- `combo(PUBKEY)` (top level only): an alias for the collection of `pk(PUBKEY)`, `pkh(PUBKEY)`, `wpkh(PUBKEY)`, and `sh(wpkh(PUBKEY))`.
+- `multi(k,PUBKEY_1,PUBKEY_2,...,PUBKEY_n)` (anywhere): k-of-n multisig script.
+- `addr(ADDR)` (top level only): the script which ADDR expands to (note that there is no address for P2PK outputs).
+- `raw(HEX)` (top level only): the script whose hex encoding is HEX.
+
+Public keys:
+- Hex encoded public keys (66 characters starting with `02` or `03`, or 130 characters starting with `04`).
+  - Inside `wpkh` and `wsh`, only compressed public keys are permitted.
+  - WIF encoded private keys may be specified instead of the correspondong public key, with the same meaning.
```

Comment:
Typo: `correspondong`

Reply:
Good catch! I can't believe I missed this typo yesterday. Perhaps it was not present then? :-)

Reply:
we definitely want to fix this one before merge 🤣

Reply:
Fixed.